### ISSUE TITLE
Use HTML button element for AccessibleButton?

### ISSUE THREAD

**Issue by luixxiul** [vector-im/element-web#22423]
### Steps to reproduce

Per https://github.com/matrix-org/matrix-react-sdk/pull/8737#discussion_r887867501

While `AccessibleButton` has been implemented with div tag since more than 6 years from [its beginning](https://github.com/matrix-org/matrix-react-sdk/blame/86276450f64c41ffd8d71bc14a758be0c4b19e37/src/components/views/elements/AccessibleButton.js#L47) and worked, currently it requires to add `element='button'` to display the button as a `button` HTML element ([example](https://github.com/matrix-org/matrix-react-sdk/blob/7c57680b93db3afeb9f52a65a5302d66d79a56b9/src/components/views/location/ShareType.tsx#L54)), which does not look so intuitive.

However, though it should be logical to replace div with button, the button with div works without an issue, and I am not quite sure how much merit it would have over possible risk for breaking the functions.

#### What did you expect?

`AccessibleButton` without element value should be displayed as `button` element instead of `div`.

#### What happened instead?

It is displayed as `div` element, whether it has attributes for accessibility, etc.

### Application version

develop branch

### Will you send logs?

No

**Comment by t3chguy** [comment]
Wouldn't work in certain places

e.g
![image](https://user-images.githubusercontent.com/2403652/172113136-60a4de85-dd21-42fb-a708-2f11233f6eb1.png)

**Comment by luixxiul** [comment]
Closing as changing does not improve something.

**Comment by t3chguy** [comment]
Nope, `button` is still important for semantics. E.g. we cannot use RTL's `isDisabled()` on `div role=button`
